academy/exchange/cloud/backend.py

```diff
@@ -461,15 +574,90 @@ def _agent_key(self, uid: EntityId) -> str:
     def _queue_key(self, uid: EntityId) -> str:
         return f'queue:{uid.uid}'
 
+    def _share_key(self, uid: EntityId) -> str:
+        return f'share:{uid.uid}'
+
+    async def share_mailbox(
+        self,
+        client: ClientInfo,
+        uid: EntityId,
+        group_uid: str,
+    ) -> None:
+        """Share a mailbox with a Globus group.
+
+        Args:
+             client: Client making the request.
+             group_uid: Group id to share mailbox with.
+             uid: Target Mailbox for sharing
+        """
+        if not await self._has_mailbox_ownership(client, uid):
+            raise ForbiddenError(
+                'Mailbox sharing requires ownership',
+            )
+        if group_uid not in client.group_memberships:
+            raise ForbiddenError(
+                f'Owner does not belong to the group:{group_uid}',
+            )
+
+        await self._client.sadd(self._share_key(uid), group_uid)  # type: ignore[misc]
+
+    async def get_mailbox_shares(
+        self,
+        client: ClientInfo,
+        uid: EntityId,
+    ) -> list[str]:
+        """Get list of globus groups the mailbox is shared with.
+
+        Args:
+             client: Client making the request.
+             uid: Mailbox id to get sharing info
+        """
+        owner = await self._client.get(
+            self._owner_key(uid),
+        )
+        if not owner:
+            raise BadEntityIdError(uid)
```

Comment:
Same question as above about what's being checked here and what the error means.